src/execution/deprecated_syscall_handler.rs

```diff
@@ -63,29 +71,79 @@ impl DeprecatedOsSyscallHandlerWrapper {
             })
             .collect();
         vm.load_data(new_segment, &retdata)?;
-        vm.insert_value((syscall_ptr + retdata_offset).unwrap(), new_segment)?;
+        vm.insert_value((syscall_ptr + retdata_offset)?, new_segment)?;
 
         Ok(())
     }
+
+    pub async fn call_contract(&self, syscall_ptr: Relocatable, vm: &mut VirtualMachine) -> Result<(), HintError> {
+        self.call_contract_and_write_response(syscall_ptr, CallContract::response_offset(), vm).await
+    }
     #[allow(unused)]
-    pub fn delegate_call(&self, syscall_ptr: Relocatable) {
-        log::error!("delegate_call (TODO): {}", syscall_ptr);
+    pub async fn delegate_call(&self, syscall_ptr: Relocatable, vm: &mut VirtualMachine) -> Result<(), HintError> {
+        self.call_contract_and_write_response(syscall_ptr, CallContract::response_offset(), vm).await
     }
-    pub fn delegate_l1_handler(&self, syscall_ptr: Relocatable) {
-        log::error!("delegate_l1_handler (TODO): {}", syscall_ptr);
+    pub async fn delegate_l1_handler(
+        &self,
+        syscall_ptr: Relocatable,
+        vm: &mut VirtualMachine,
+    ) -> Result<(), HintError> {
+        self.call_contract_and_write_response(syscall_ptr, CallContract::response_offset(), vm).await
     }
-    pub fn deploy(&self, syscall_ptr: Relocatable) {
-        log::error!("deploy (TODO): {}", syscall_ptr);
+    pub async fn deploy(&self, syscall_ptr: Relocatable, vm: &mut VirtualMachine) -> Result<(), HintError> {
+        let syscall_handler = self.deprecated_syscall_handler.read().await;
+        let mut execution_helper = syscall_handler.exec_wrapper.execution_helper.write().await;
+
+        // Advance the result iterator
+        let _ = execution_helper.result_iter.next();
```

Comment:
This returns an option, should it assert in some way that the option is `Some` since there should necessarily be something in `result_iter` at this point?